libs/klv/src/main/java/org/codice/ddf/libs/klv/data/text/KlvEncodingDetectedString.java

```diff
@@ -47,14 +47,17 @@ protected void decodeValue(Klv klv) {
     CharsetDetector charsetDetector = new CharsetDetector();
     charsetDetector.setText(bytes);
     CharsetMatch charsetMatch = charsetDetector.detect();
-    try {
-      value = new String(bytes, charsetMatch.getName());
-      return;
-    } catch (UnsupportedEncodingException e) {
-      LOGGER.trace(
-          "Unsupported encoding of %s, falling back to default encoding", charsetMatch.getName());
+
+    if (charsetMatch != null) {
+      try {
+        value = new String(bytes, charsetMatch.getName());
+      } catch (UnsupportedEncodingException e) {
+        LOGGER.trace(
+            "Unsupported encoding of %s, falling back to default encoding", charsetMatch.getName());
+      }
+    } else {
```

Comment:
Should we be using defaultCharset() or UTF-8?

Reply:
Not entirely sure, but this was what we had there before. I think that would be better addressed in another ticket.